youtube_dl/jsinterp.py

```diff
@@ -534,25 +648,30 @@ def extract_function_from_code(self, argnames, code, *global_stack):
             name = self._named_object(
                 local_vars,
                 self.extract_function_from_code(
-                    [x.strip() for x in mobj.group('args').split(',')],
+                    self.build_arglist(mobj.group('args')),
                     body, local_vars, *global_stack))
             code = code[:start] + name + remaining
         return self.build_function(argnames, code, local_vars, *global_stack)
 
     def call_function(self, funcname, *args):
         return self.extract_function(funcname)(args)
 
+    @classmethod
+    def build_arglist(cls, arg_text):
+        if not arg_text:
+            return []
+        return list(filter(None, (x.strip() or None for x in cls._separate(arg_text))))
```

Comment:
Same. The `filter` doesn't make sense to me

Reply:
Removing empty items, but there shouldn't be any:
```js
function fred(a,b,c) {
  return b;
}
fred(1, ,2);
/*
Exception: SyntaxError: expected expression, got ','
@Scratchpad/1:4
*/
function fred(a,,c) {
  return c;
}
/*
Exception: SyntaxError: missing formal parameter
@Scratchpad/1:1
*/
```
So probably just strip and let it crash elsewhere.